yang/frr-bgp-neighbor.yang

```diff
@@ -0,0 +1,139 @@
+submodule frr-bgp-neighbor {
+  yang-version 1.1;
+
+  belongs-to frr-bgp {
+    prefix "bgp";
+  }
+
+  include frr-bgp-common-structure;
+  include frr-bgp-common-multiprotocol;
+
+  organization
+    "FRRouting";
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org> FRR Development
+     List: <mailto:dev@lists.frrouting.org>";
+  description
+    "This submodule contains general data definitions for use in BGP neighbor.
+
+     Copyright 2020 FRRouting
+
+     Redistribution and use in source and binary forms, with or without
+     modification, are permitted provided that the following conditions
+     are met:
+
+     1. Redistributions of source code must retain the above copyright notice,
+     this list of conditions and the following disclaimer.
+
+     2. Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+
+     THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+     \"AS IS\" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+     LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+     A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+     HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+     SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+     LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+     DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+     THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+     (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+     OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.";
+
+  revision 2019-12-03 {
+    description
+      "Initial revision.";
+  }
+
+  grouping neighbor-group-config {
+    container neighbor-group-config {
```

Comment:
This is for peer-groups, I take it? It looks like it's the actual peer's config, in which case I'm not understanding why it's called `-group-config`. Elsewhere in the model I see a `peer-group` name leaf within the neighbor definition, and then in the `peer-group` container it looks like most of this config is duplicated. How does the inheritance between peer-group and neighbor work?

Reply:
I have removed the container and renamed the grouping to neighbor-parameters. 
```
Neighbor --> has It's own config
PeerGroup --> has it's own config

 router bgp 10
 bgp router-id 10.10.0.1
 no bgp default ipv4-unicast
 bgp bestpath compare-routerid
 neighbor IBGP peer-group
 neighbor IBGP remote-as 10
 neighbor IBGP update-source 10.10.0.1
 neighbor IBGP advertisement-interval 1
 neighbor IBGP timers connect 10
 neighbor 10.10.0.2 peer-group IBGP
 neighbor 10.10.0.3 peer-group IBGP

PeerGroup IBGP is configured.
Now neighbor 10.10.0.2 has a leaf-ref to PeerGroup IBGP.

Once this is given, at the backend one can modify the peer mapping inside peer_group
       /* Peer-group client list. */
        struct list *peer;  

In this way inheritance between peer-group and neighbor is maintained.

```